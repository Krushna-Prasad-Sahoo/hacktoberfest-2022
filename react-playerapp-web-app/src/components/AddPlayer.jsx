import React, { Component } from 'react'
import {PlayerContext} from '../data/PlayerContext';
import axios from 'axios';
//import ShowPlayer from './ShowPlayer';
// import Player from '../../Players.json'
export default class AddPlayer extends Component {
      //  declare state with required properties
      constructor(props){
        super(props);
        this.state={
            pName:"",
            pCountry:"",
            matchPlayed:""
        }
      }
      static contextType = PlayerContext
     submitHandler = async (e)=>{
      e.preventDefault();
      const user = this.context
      let data = {
        name:this.state.pName,
        country:this.state.pCountry,
        matchesPlayed:this.state.matchPlayed
      }
      const res = await axios.post('http://localhost:3000/players', data);
      console.log(res);

      // user[0].push(data);
      // console.log("user1" , user)
      // return <ShowPlayer user={user[0]}/>
      window.location.href = "/showplayer"
      // this.props.history.push("/showplayer")
      }
    //  submitHandler(e){
      
    
    //   // declare and initialise players ,setPlayers from context 
    //   // use axios.post to send new player data to json api
    //   // and store the same to context also using setPlayer 

    //   // reset the text fields and show acknowledgement
      
    // }
    render() {

        return (
          <div>
             {/* display form to accept name,country and matchesPlayed  of a player */}
             <form onSubmit={this.submitHandler}>
               <div className='add'>
               <div>
            <input type="text" placeholder='playername' value={this.state.pName}  onChange={(e) => this.setState({pName:e.target.value})}/>
            </div>
            <div>

           
            <input type="text" placeholder='Country' value={this.state.pCountry} onChange={(e) => this.setState({pCountry:e.target.value})} />
            </div>

            <div>
            <input type="text" placeholder='matchPlayed' value={this.state.matchPlayed} onChange={(e) => this.setState({matchPlayed:e.target.value})} />
            </div>
            <div className='sub'>
            <button>Submit</button>
            </div>
            
            
            </div>
             </form>
            </div>
        )
    }
}
AddPlayer.contextType=PlayerContext;
