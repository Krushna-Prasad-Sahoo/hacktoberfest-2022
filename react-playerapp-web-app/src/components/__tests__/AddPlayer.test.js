 
import AddPlayer from '../AddPlayer';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, render, mount } from 'enzyme';
xdescribe("Add Player Testing",()=>{
    const wrapper = shallow(<AddPlayer/>) 
it('renders correctly Addplayer', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Must contain text field to accept playername',()=>{
      expect(wrapper.contains(
      <label for="playername">Player Name</label>
    )).toBeTruthy();
    
  }) 
  it('form must contain textfield with id matchesPlayed',()=>{
    expect(wrapper.find('input#matchesPlayed')).toBeTruthy();
  })
  it('form must contain a submit button with caption Add Player',()=>{
    expect(wrapper.contains(<button type="submit" class="btn btn-primary"  >Add Player</button>)).toBeTruthy();  
  })
  it('form must contain a textfield with id country',()=>{
      expect(wrapper.find('input#country')).toBeTruthy();
  })
  it('form must contain a textfield with id playername',()=>{
      expect(wrapper.find('input#playername')).toBeTruthy();
  })
//   it('should fail if no credentials are provided', () => {
//     const fakeEvent = { preventDefault: () => console.log('preventDefault') };

//     expect(wrapper.find('form#addplayer')).toBeTruthy();
//     wrapper.find('form').simulate('submit', fakeEvent);
//     // expect(wrapper.find(Notification).length).toBe(1);
// });
})
