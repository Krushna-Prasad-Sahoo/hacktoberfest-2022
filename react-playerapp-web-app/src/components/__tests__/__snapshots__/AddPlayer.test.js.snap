// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Player Testing renders correctly Addplayer 1`] = `
<div>
  <form
    id="form-addplayer"
    onSubmit={[Function]}
  >
    <div
      class="form-group"
    >
      <label
        for="playername"
      >
        Player Name
      </label>
      <input
        class="form-control"
        id="playername"
        onChange={[Function]}
        placeholder="Enter Player Name"
        required={true}
        type="text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="country"
      >
        Country
      </label>
      <input
        class="form-control"
        id="country"
        onChange={[Function]}
        placeholder="Enter Country Name"
        required={true}
        type="text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="matchesPlayed"
      >
        Total Matches Played
      </label>
      <input
        class="form-control"
        id="matchesPlayed"
        onChange={[Function]}
        placeholder="Enter Total Matches Played"
        required={true}
        type="text"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Add Player
    </button>
  </form>
</div>
`;
